### DISCUSSION TITLE
Proper instructions on upgrading to latest version

### DISCUSSION THREAD

**Discussion by bingston** [awawa-dev/HyperHDR#813] [Q&A]
Forgive the silly question - I'm fairly new to all of this.  The tutorial that I followed installed version 19 but I see that version 20 is available now.  What's the best way to upgrade to the latest version?  

The original command I used to install version 19 was `wget https://github.com/awawa-dev/HyperHDR/releases/download/v19.0.0.0/HyperHDR-19.0.0.0-Linux-aarch64.deb`.  That was followed by `sudo apt -y install ./HyperHDR-19.0.0.0-Linux-aarch64.deb`. 

I didn't know if it was as simple as just replacing the "19" with "20" essentially.  I was worried about both versions running and causing conflicts so I wanted to ask here to get the proper instructions.

Love this project so far and things are looking great!  Thanks so much for everyone's work here!

**Comment by awawa-dev** [comment]
Hi
Thanks. You can remove HyperHDR using this command (the configuration will not be deleted): `sudo apt remove hyperhdr`
I think what you want to do should work. You can also use APT repository to install HyperHDR automatically https://github.com/awawa-dev/HyperHDR/wiki/Installation#debian-and-ubuntu-apt-repository

**Reply by bingston** [reply]
Thanks so much!  I appreciate it

**Comment by thepuncake** [comment]
Hi there, I did this `sudo apt update && sudo apt install hyperhdr -y` on a hyperhdr proxmox lxc and very sadly lost my configuration. Could there be a way to get it back? Thanks.

**Reply by awawa-dev** [reply]
You haven't lost your configuration but I guess your old setup ran on the different user.

**Comment by thepuncake** [comment]
That's strange as I ran the update command from within the lxc container. Otherwise I'm always logged in as root. Any thoughts on how I could retrieve it? I mostly care about my custom led layout and having to redesign that would be an absolute pain.

Edit: If it helps I installed hyperhdr through tteck proxmox scripts.

**Reply by awawa-dev** [reply]
Check what user is used for HyperHDR service now. Your old config should be present in a home directory in .hyperhdr (may be hidden) folder of the user who previously was used for HyperHDR service. Setup doesn't remove your old config under any circumstances.

**Comment by thepuncake** [comment]
Upon checking there is no other user in the system except root@pam. In the task history that's all I see too. When I check the .hyperhdr directory (as root) it contains a db file which is the newly created one and the lut tables. Excuses if I'm missing something but I'm baffled as to why my config isn't there so any guidance would be appreciated. 😊

**Reply by awawa-dev** [reply]
V20 always creates a backup (also in home .hyperhdr db folder) of database if the old version eg from V19 was present. If there was no backup then there was no DB earlier at least for this user.

BTW Unfortunately can't help with custom proxmox scripts as I don't maintain nor support them.

**Comment by thepuncake** [comment]
Thanks for your replies and attempting to help me. Sadly I still don't know what to do to retrieve my config or where to go from here. Btw I wasn't accusing hyperhdr or you in any way for the missing config, I'm just trying to solve this if I can.

**Reply by awawa-dev** [reply]
Maybe proxmox messed up with the user configuration? Unfortunately I dont know what was you home folder before nor I use proxmox. Try to search for a file `hyperhdr.db` on the host and inside the proxmox container.

**Reply by awawa-dev** [reply]
something like `find / -name hyperhdr.db`
you must do it on both environments because from what I guess proxmox may probably join/synchronize user home folder (not sure about it). So the config may be here or there.

**Reply by thepuncake** [reply]
Thanks for thinking along, however the answer is probably this one when it comes to the aforementioned script. Perhaps it helps others.
![Screenshot_20240519-211009](https://github.com/awawa-dev/HyperHDR/assets/5490377/f3ed7dee-b01a-4bdf-a51b-35f15775b7ee)

**Reply by awawa-dev** [reply]
That's not true, the user configuration file is not a part of the installer nor the part of the uninstaller script. It is created when you start the app for the first time (for each user). You can uninstall and install the app any time you want on standard the OS. Cannot guarantee proxmox behavior (did it reset the whole container for some reason???) or it's system user configuration of the proxmox , it's not supported by me.

**Reply by thepuncake** [reply]
I'm not sure if it resets the whole container, maybe would be good for tteck to fly over and clarify. I'll try to let him know or is there a way to tag him? Can't seem to make it work.

**Reply by awawa-dev** [reply]
I dont know. Here is the uninstaller script for deb packages: https://github.com/awawa-dev/HyperHDR/blob/master/cmake/debian/prerm

**Reply by thepuncake** [reply]
Ok, thanks for trying to help, seems the problem is for sure proxmox related. Here I go setting things up again, will remember to backup this time heh.

**Reply by Henchman-21** [reply]
Just a note, I just had this problem though going from v19->v20 last night.  Ran the update w/ the -y switch and it looks like instead of installing/updating my .hyperhdr under the pi user account, it ran and installed in /root as well.  Now I can't get to the service via web url, even though it says it's running.

pi@HyperHDR:/usr/bin $ sudo find / -name hyperhdr.db
/root/.hyperhdr/db/hyperhdr.db
/home/pi/.hyperhdr/db/hyperhdr.db

Going to pull the db off to backup and try and salvage, but curious it happened.  I've upgraded pretty much the same way since v15 or so.  I'm running off a pi w/ ubuntu.

**Reply by awawa-dev** [reply]
1) first, how to start/stop services and debug what's going on https://github.com/awawa-dev/HyperHDR/wiki/Installation#Linux
2) how to switch user: , https://github.com/awawa-dev/HyperHDR/wiki/FAQ usually it is done the other way pi -> root, apt update will not restore service as a root user, so this is quite a common problem for users who have an unsupported ws281x strip driven directly from rpi which requires root

>  I've upgraded pretty much the same way since v15 or so. 

there was no APT repo then...

**Reply by Henchman-21** [reply]
Yeah, idk.  I ended up killing/disabling the hyperhdr@pi service and everything's back to normal (on v20).  Either way, love the software, appreciate everything you do!

**Comment by versionDefect** [comment]
> Hi
> 
> Thanks. You can remove HyperHDR using this command (the configuration will not be deleted): `sudo apt remove hyperhdr`
> 
> I think what you want to do should work. You can also use APT repository to install HyperHDR automatically https://github.com/awawa-dev/HyperHDR/wiki/Installation#debian-and-ubuntu-apt-repository

Hi. I've got HHDR running on a Pi 3B. Works well but an update wouldn't hurt. 

What's the best upgrade process for a Raspberry Pi?

**Reply by awawa-dev** [reply]
Hi
Depending on the operating system version. If you have a Bullseye based OS, you can upgrade via the APT repository or flash a new Bullseye HyperHDR v20 SD image after backing up your configuration (web panel v17 and above -> general tab, backup is always recommended before updating). If you want to update your system using official images, be careful because there is a kernel bug that affects Rpi3 and occurs in the OS Bookworm 6.6 kernel: https://github.com/awawa-dev/HyperHDR/discussions/848 (kernel downgrade required to fix it)

**Comment by Pastaloverzzz** [comment]
Installed hyperhdr on debian(raspberry pi) and my configuration is gone as well after updating from version 20 to 21.

**Reply by awawa-dev** [reply]
led driver?

**Reply by Pastaloverzzz** [reply]
raspberry pi is my led driver. I found my backup .json file on there on /home/user/.hyperhdr/db
Do you know how i can put it back?

**Reply by awawa-dev** [reply]
rpi GPIO PWM driver?

**Reply by Pastaloverzzz** [reply]
Yes

**Reply by awawa-dev** [reply]
Well, using official unsupported drivers can lead you astray even where you wouldn't expect it. To calm you down, no configuration was lost/overwritten (at least not by the HyperHDR installer), only the user the service is running under has most likely changed -> so has the home directory with the HyperHDR configuration. Rpi PWM requires the root user. How to change the service back to root can be found in Wiki/FAQ. And if you really want to restore some unknown json backup, you can do it through the interface... but make sure it's not an old configuration.

**Reply by andyjb22** [reply]
I just had a similar issue, followed a guide for reinstallation and a simple modification to the service file brought it all bak to operate as root

sudo nano /etc/systemd/system/hyperhdr@.service

Remove the line 

user = %i

Save the file

On reboot my config had returence

**Reply by awawa-dev** [reply]
With this workaround you damage the proper service script for systemd as it needs the user name passed as an argument. So enable and run the service on the proper user instead as it's described in the wiki/FAQ.

**Reply by andyjb22** [reply]
I will look through it, thanks

**Reply by andyjb22** [reply]
I managed to follow through the threads to get persistent root operation, the workaround helped get my config downloaded as a back up which made it easier to import once I repaired the system file and adjusted set it to run as root.

added thread to comment to make it easy to find form this thread

https://github.com/awawa-dev/HyperHDR/discussions/43

Thanks for getting back to me, its a great bit of software! Appreciate your hard work!
